crates/shared/src/price_estimation/competition.rs

```diff
@@ -50,20 +50,23 @@ type PriceEstimationStage<T> = Vec<(String, T)>;
 /// stage Returns a price estimation early if there is a configurable number of
 /// successful estimates for every query or if all price sources returned an
 /// estimate.
-pub struct RacingCompetitionEstimator<T> {
+pub struct RacingCompetitionEstimator<T, N> {
     inner: Vec<PriceEstimationStage<T>>,
     successful_results_for_early_return: NonZeroUsize,
+    native: N,
 }
 
-impl<T: Send + Sync + 'static> RacingCompetitionEstimator<T> {
+impl<T: Send + Sync + 'static, N: Send + Sync + 'static> RacingCompetitionEstimator<T, N> {
     pub fn new(
         inner: Vec<PriceEstimationStage<T>>,
         successful_results_for_early_return: NonZeroUsize,
+        native: N,
```

Comment:
nit: can be option to not use a Noop estimator and make the 1:1 exchange rate for the native token estimates explicit?